src/map/unit.cpp

```diff
@@ -2965,15 +2965,15 @@ static int32 unit_attack_timer_sub(struct block_list* src, int32 tid, t_tick tic
 	}
 
 	if( !battle_check_range(src,target,range) ) {
-	  	// Within range, but no direct line of attack
-		if( ud->state.attack_continue ) {
-			if(ud->chaserange > 2)
-				ud->chaserange-=2;
-
+		// Within range, but no direct line of attack
+		// This code can usually only be reached if OFFICIAL_WALKPATH is disabled
+		if (ud->state.attack_continue && ud->chaserange > 1) {
+			ud->chaserange = std::max(1, ud->chaserange - 2);
 			unit_walktobl(src,target,ud->chaserange,ud->state.walk_easy|2);
```

Comment:
I am not sure, if we should respect ud->state.walk_easy here.
Obviously we are standing in front of an obstacle.
Using easy path search will not walk around the obstacle, which sounds like an infinite loop to me again.
So maybe we only should start walking, if walk_easy == 0 and otherwise give up?
Maybe it should be added to the conditions of the if around this block?

Reply:
I think it would be weird if a unit that's supposed to only walk in a straight line would suddenly start using complex pathfinding.

But we should do as you commented in the previous comment - return 0 when walkpath fails.

There can't be an endless loop anymore because if the walkpath fails it will never trigger the "reached end of walkpath" event again.

Reply:
Why I am thinking of an infinite loop here:
You are already in attack range (variable name "range" here).
You try to walk chaserange - 2 cells close to the target.
But I think in quite a few scenarios, you will already stand inside chaserange - 2 cells distance to the target.

Do you not think that this could loop as well?

Reply:
We give ud->chaserange back into unit_walktobl which defines the range at which it stops next.

So let's say you are at distance 3 and ud->chaserange is 5. Then we set ud->chaserange to 3 and start chasing again -> it immediately finds out it's already in attack range, stops and comes back here.
Now it's at distance 3 and ud->chaserange is 3. We set ud->chaserange to 1 and start chasing again.
And if it still fails at ud->chaserange=1, we quit.